src/harness/reference_models/interference/interference.py

```diff
@@ -413,57 +394,73 @@ def computeInterferenceFssCochannel(cbsd_grant, constraint, fss_info, max_eirp):
 def getFssMaskLoss(cbsd_grant, constraint):
   """Gets the FSS mask loss for a FSS blocking protection constraint."""
   # Get 50MHz offset below the lower edge of the FSS earth station
-  offset = constraint.low_frequency - 50.e6
-
+  offset = constraint.high_frequency - 50*MHZ
   # Get CBSD grant frequency range
   cbsd_freq_range = cbsd_grant.high_frequency - cbsd_grant.low_frequency
+  # Define the initial attenuation values to 0
+  fss_mask_attenuation = 0
+  fss_mask_attenuation_seg1 = 0
+  fss_mask_attenuation_seg2 = 0
+  # If the grant overlap frequency lies between fp1-50 MHz to fp1, where fp1 is the lower edge of the FSS's passband
+  # Segment 1 calculation is applied
+  if ( offset <= cbsd_grant.low_frequency and
+         constraint.high_frequency >= cbsd_grant.high_frequency):
+    # Calculate the total of 1MHZ frequencies present in the grant frequency range
+    num_one_mhz_bin_seg1 = cbsd_freq_range/int(MHZ)
+    # Calculate the fss_mask for every 1MHZ frequency
+    for i in range (0, num_one_mhz_bin_seg1 ):
+      fss_mask_attenuation +=10**(-(((constraint.high_frequency - cbsd_grant.low_frequency)/int(MHZ)) - 0.5 -i)*0.6 - 0.5)/10
+    # Calculate the average attenuation value per MHZ
+    fss_mask_attenuation_seg1 = fss_mask_attenuation/num_one_mhz_bin_seg1
```

Comment:
I am not sure why fss_mask_attenuation is divided by the channel size, unless the function returns loss per MHz. If that is the case, please document the function. However, since for blocking protection we need to use total amount of power (not per MHz), I do not see a reason to calculate loss per MHz.

Reply:
We are calculating attenuation due to FSS filter in the center of each 1 MHz bin and average it over the bandwidth of grant. The per MHz computation is to only take into account filter loss at each MHz bin but is then aggregated over grant bandwidth. At the end we get FSS filter's contribution to the grant bandwidth and not on a per MHz basis.